.github/scripts/crib/lib/check-route53-records.js

```diff
@@ -0,0 +1,83 @@
+import {
+  Route53Client,
+  ListResourceRecordSetsCommand,
+} from "@aws-sdk/client-route-53";
+
+// us-east-1 is the global region used by Route 53.
+const route53Client = new Route53Client({ region: "us-east-1" });
+
+// Function to wait for a specified amount of time (ms).
+const wait = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+/**
+ * Check if Route 53 records exist for a given Route 53 zone.
+ *
+ * @param {string} hostedZoneId The ID of the hosted zone.
+ * @param {string[]} recordNames An array of record names to check.
+ * @param {number} maxRetries The maximum number of retries.
+ * @param {number} initialBackoff The initial backoff time in milliseconds.
+ * @returns {Promise<boolean>} True if records exist, false otherwise.
+ */
+async function route53RecordsExist(
+  hostedZoneId,
+  recordNames,
+  maxRetries = 8,
+  initialBackoff = 2000 // 2 seconds
+) {
+  let attempts = 0;
+
+  console.info("Checking DNS records in Route 53...");
+  while (attempts < maxRetries) {
+    try {
+      let isTruncated = true;
+      let nextRecordName;
+      let nextRecordType;
+      let allRecordSets = [];
+
+      while (isTruncated) {
```

Comment:
It would be worth wrapping this in a function that indicates that it is a paginator. You might be able to actually make your own custom paginator rather than rolling your own if you're interested in trying. See https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-route-53/src/pagination/ListCidrBlocksPaginator.ts

In this case, the page markers would be "isTruncated" along with "maxResults"

Reply:
Tried to address in here: 44a8083cec4bc16cef3ff799b8158fd6f9a4118a.